### ISSUE TITLE
True strain and engineering strain; the possibility of missing coefficient 2

### ISSUE THREAD

**Issue by Alirezarei** [idaholab/moose#26150]
I was considering the ComputeElasticityTensor for an orthotropic material and I realized that the C44, C55, and C66 are considered to be shear modulus G12, G31, G23. Please refer to this link: https://mooseframework.inl.gov/source/materials/ComputeElasticityTensor.html
If this is the case, the shear strains calculated using ADComputeSmallStrain should be doubled, but it seems that the calculated shear strains are the engineering strain that are being multiplied to shear modulus to give us the shear stress. Therefore, I believe a "2" coefficient is missing.

In other words, if we are calculating the shear stress we need to use the following formulae: τ_xy=2Gε_xy
The shear modulus in the C44, C55, and C66 members of the elasticity tensor calculated by "ComputeElasticityTensor" are considered to be shear modulus G12, G31, G23. Considering that the strains calculated in ADComputeSmallStrain are engineering strains and the multiplication of elasticity tensor by strains will give us the stresses, a factor "2" is missing in the equations for shear stress calculations.

**Comment by lindsayad** [comment]
@bwspenc @dschwen @jiangwen84 can someone look at this?

**Comment by Alirezarei** [comment]
Hi,
Any updates on this?

Thank you.

**Comment by sapitts** [comment]
Hi @Alirezarei

The coefficient of `2` that you describe is common for the Voigt notation. Voigt notation is used in mechanics to represent the Rank-2 stress and strain tensors as vectors (of size 6) and to represent the Rank-4 stiffness tensor as a 6x6 matrix. In this representation, the factor of 2 is necessary for the shear stress-shear strain relationship to account for the sum of squares, see [https://en.wikipedia.org/wiki/Voigt_notation](https://en.wikipedia.org/wiki/Voigt_notation) for more details and equations.

In MOOSE's tensor mechanics, we store the stress and strain tensors as the full Rank-2 tensor and the stiffness tensor as the full Rank-4 tensor, with 81 components (hence the name of this MOOSE module), see the short paragraph on the documentation page you linked right above Eq. (2). The constitutive elasticity relationship is calculated as

$$ \sigma_{ij} = C_{ijkl}\epsilon_{kl} $$

and both the small strain and finite strain formulations calculate the Cauchy, or true stress, when the default tensor mechanics kernels are used, as in the case of `ADComputeSmallStrain`.

You can see how the Rank-4 stiffness tensor is built for the various symmetric cases in the code: [RankFourTensor.h](https://github.com/idaholab/moose/blob/next/framework/include/utils_nonunity/RankFourTensor.h)

Hope this helps
